packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -72,67 +74,92 @@ type CurrentUIEvent = {
 const PUBLISHED_EVENTS = new Set<string>();
 type ALChannel = Channel<ALChannelUIEvent>;
 
+interface UIEventConfig {
+  readonly eventName: keyof GlobalEventHandlersEventMap,
+  // A callable filter for this event, returning true if the event should be emitted, or false if it should be discarded up front
+  eventFilter?: (domEvent: GlobalEventHandlersEventMap[UIEventConfig['eventName']]) => boolean;
+  interactableElementsOnly?: boolean;
+  cacheElementReactInfo: boolean;
+}
+
 export type InitOptions = Types.Options<
   ALSharedInitOptions &
   {
-    uiEvents: Array<string>;
+    uiEvents: Array<UIEventConfig>;
     channel: ALChannel;
-    cacheElementReactInfo: boolean;
   }
 >;
 
 export function publish(options: InitOptions): void {
-  const { uiEvents, flowletManager, channel, domSurfaceAttributeName, cacheElementReactInfo } = options;
+  const { uiEvents, flowletManager, channel, domSurfaceAttributeName} = options;
 
 
   const newEventsToPublish = uiEvents.filter(
-    event => !PUBLISHED_EVENTS.has(event),
+    event => !PUBLISHED_EVENTS.has(event.eventName),
   );
 
-  trackInteractable(newEventsToPublish);
+  trackInteractable(newEventsToPublish.map(ev => ev.eventName));
 
   let lastUIEvent: CurrentUIEvent | null;
   let flowlet: ALFlowlet;
   const defaultTopFlowlet = new flowletManager.flowletCtor("/");
 
-  newEventsToPublish.forEach(eventName => {
+  newEventsToPublish.forEach((eventConfig => {
+    const {eventName, eventFilter, cacheElementReactInfo, interactableElementsOnly = true } = eventConfig;
     // Track event in the capturing phase
-    window.document.addEventListener(eventName, (event: Event) => {
-      const element = getInteractable(event.target, eventName);
+    window.document.addEventListener(eventName, (event: GlobalEventHandlersEventMap[typeof eventConfig.eventName]) => {
```

Comment:
the type of `event` won't further narrow down because `eventName` is string. That kind of thing only works if you have generics. You might as well just keep it as `MouseEvent | KeyboardEvent`